src/language_server_transport.rs

```diff
@@ -33,6 +34,7 @@ pub fn start(
         .stdin(Stdio::piped())
         .stdout(Stdio::piped())
         .stderr(Stdio::piped())
+        .current_dir(root)
```

Comment:
For me it fails when I am for example in the directory of 1 project, then open a file in another project that's at the same level of the filesystem or higher. This second one will never get LSP support. With this change that does work i.e. the lsp starts in the root of the second project for that newly opened file. 
Does that make sense?

Reply:
if i look for `rootUri` the only line I find is https://github.com/OmniSharp/omnisharp-roslyn/blob/ac7b9b8509356e39583de2b9fdf363005e6c8595/src/OmniSharp.LanguageServerProtocol/LanguageServerHost.cs#L201

Where the first check for whether application.Root is null or empty, if not they use that over the starting parameters. Maybe I should open a PR there as well...

Reply:
> For me it fails when I am for example in the directory of 1 project, then
> open a file in another project that's at the same level of the filesystem
> or higher. This second one will never get LSP support.

Thanks, I can reproduce now.
I copied the same project into `./examples-csharp` and `../examples-csharp`.
Then I do

	kak {.,..}/examples-csharp/MutexSingleInstanceAndNamedPipe/MutexSingleInstanceAndNamedPipe/FormMain.cs

Two language servers are started but only the first one works.

We send this in `initialize`

```
    "rootPath": "$PWD/../examples-csharp/MutexSingleInstanceAndNamedPipe",
    "rootUri": "file://$PWD/../examples-csharp/MutexSingleInstanceAndNamedPipe",
    "workspaceFolders": [
      {
        "name": "$PWD/../examples-csharp/MutexSingleInstanceAndNamedPipe",
        "uri": "file://$PWD/../examples-csharp/MutexSingleInstanceAndNamedPipe"
      }
    ]
```


but it looks like Omnisharp ignores that when `$PWD` is already inside a project.

Haven't done much research but this seems like a bug in Omnisharp.

I wonder if there is another LSP client that sets the LSP server process directory; it would be interesting to find out why.

> Where the first check for whether application.Root is null or empty, if not they use that over the starting parameters.

Yeah that sounds like the exact place where things go wrong.
`rootUri` should take precedence over.

Looks like this (inadvertently?) regressed in 30db3e09 (Updated host to use provided solution file., 2021-02-28) from [this PR](https://github.com/OmniSharp/omnisharp-roslyn/pull/2099)
Before that commit, `initializeParams.RootUri` was always used if non-null.
After that commit, `application.ApplicationRoot` was always used if non-null, potentially overriding a non-null `initializeParams.RootUri`.
My guess is that this was to support the `omnisharp -s` switch to override anything else.
However that doesn't make sense if no `-s` was specified. In this case `application.ApplicationRoot` defaults to `cwd`; it should not override `initializeParams.RootUri`.

Are you able to work on a PR for `omnisharp`?

Reply:
Yes, I think I'll do that! Thanks for your research already, that helps hopefully to convince them to revert/adapt the behavior if I don't find out the most elegant way to solve this myself. I'll try anyway

Reply:
I've reported it upstream as https://github.com/OmniSharp/omnisharp-roslyn/issues/2613
Since this seems to be a regression, it's probably an easy fix for them